buffer/src/main/java/io/netty/buffer/PoolChunkList.java

```diff
@@ -49,6 +51,8 @@
         this.minUsage = minUsage;
         this.maxUsage = maxUsage;
         maxCapacity = calculateMaxCapacity(minUsage, chunkSize);
+        freeMinThreshold = (maxUsage == 100) ? 0 : (int) (chunkSize * (100L - maxUsage + 1) / 100L);
+        freeMaxThreshold = (minUsage == 100) ? 0 : (int) (chunkSize * (100L - minUsage + 1) / 100L - 1);
```

Comment:
@netudima nit: can you add a comment to explain the "formula" you used to calculate both ?

Reply:
Added a comment + updated the logic a bit to fully align old usage() based logic and the new one. How it has been verified:  https://gist.github.com/netudima/454ba97d8154109a50ae831bafc8e151